game-core/src/main/java/games/strategy/debug/error/reporting/ErrorReportComponents.java

```diff
@@ -1,58 +1,61 @@
 package games.strategy.debug.error.reporting;
 
 import java.awt.Component;
-import java.util.function.Consumer;
+import java.util.function.BiConsumer;
 import java.util.function.Supplier;
 
 import javax.swing.JButton;
+import javax.swing.JFrame;
 import javax.swing.JTextArea;
+import javax.swing.JTextField;
 
 import com.google.common.annotations.VisibleForTesting;
 
 import lombok.Builder;
-import swinglib.ConfirmationDialogBuilder;
+import swinglib.DialogBuilder;
 import swinglib.JButtonBuilder;
 import swinglib.JTextAreaBuilder;
+import swinglib.JTextFieldBuilder;
 
 
 class ErrorReportComponents {
 
   @VisibleForTesting
   enum Names {
-    ERROR_DESCRIPTION, ADDITIONAL_INFO_NAME, UPLOAD_BUTTON, PREVIEW_BUTTON
+    DESCRIPTION, TITLE, UPLOAD_BUTTON, PREVIEW_BUTTON
   }
 
   @Builder
   static class FormHandler {
-    private final Consumer<UserErrorReport> guiDataHandler;
+    private final BiConsumer<JFrame, UserErrorReport> guiDataHandler;
     private final Supplier<UserErrorReport> guiReader;
   }
 
   JTextArea descriptionArea() {
     return JTextAreaBuilder.builder()
-        .componentName(Names.ERROR_DESCRIPTION.toString())
+        .componentName(Names.DESCRIPTION.toString())
         .columns(10)
         .rows(2)
         .build();
   }
 
-  JTextArea additionalInfo() {
-    return JTextAreaBuilder.builder()
-        .componentName(Names.ADDITIONAL_INFO_NAME.toString())
+  JTextField titleField() {
+    return JTextFieldBuilder.builder()
+        .componentName(Names.TITLE.toString())
         .columns(10)
-        .rows(2)
+        .maxLength(40)
         .build();
   }
 
 
-  JButton createSubmitButton(final FormHandler config) {
+  JButton createSubmitButton(final JFrame jframe, final FormHandler config) {
```

Comment:
Nit: The `jframe` variable name is a bit inconsistent.  We don't typically call `JButton`s `jbutton`, `JTextField`s `jtextField`, etc.